internal/hcsoci/create.go

```diff
@@ -193,6 +193,14 @@ func CreateContainer(ctx context.Context, createOptions *CreateOptions) (_ cow.C
 			log.G(ctx).WithError(err).Debug("failed createHCSContainerDocument")
 			return nil, resources, err
 		}
+		// TODO dcantah: Not too satisfied with how this is added to resources here
+		// but its tricky.
```

Comment:
Did you mean to keep this comment in? If so, can you elaborate instead on why it's tricky

Reply:
I did but looking now I dont necessarily think it needs to stay. I'll probably remove when fixing the above comments but the gist of it is that it wouldn't be ideal to add it to resources in allocateWindowsResources like the others as we'd need to add a new return value to return the instance(s) along with the resources so we can add it to the doc in createWindowsContainerDocument. Creating the doc function would then need a new parameter so it could be passed in (unless we'd wanna call hcs to query the instances a second time which I'm going to go with no on this haha).